netif.c

```diff
@@ -39,16 +40,16 @@ void ndn_netif_auto_add(void)
     }
 
     /* get list of interfaces */
-    kernel_pid_t ifs[GNRC_NETIF_NUMOF];
-    size_t ifnum = gnrc_netif_get(ifs);
+    size_t ifnum = gnrc_netif_numof(); //gnrc_netif_get(ifs);
 
     if (ifnum == 0) {
         DEBUG("ndn: no interfaces registered, cannot add netif\n");
         return;
     }
 
-    for (int i = 0; i < GNRC_NETIF_NUMOF; ++i) {
-        kernel_pid_t iface = ifs[i];
+    int i = 0;
+    for (gnrc_netif_t* it = gnrc_netif_iter(NULL) ; it != NULL ; it = gnrc_netif_iter(it), i++) {
```

Comment:
this can be simplified like in [here](https://github.com/RIOT-OS/RIOT/blob/8a6d04ce52b264250b8e8c0f16b918eb38053c71/examples/gnrc_minimal/main.c#L35)